transport/tchannel/channel_outbound.go

```diff
@@ -155,11 +155,14 @@ func (o *ChannelOutbound) Call(ctx context.Context, req *transport.Request) (*tr
 	if err != nil {
 		return nil, toYARPCError(req, err)
 	}
-
+	reqHeaders := req.Headers.Items()
+	if o.transport.exactCaseHeader {
+		reqHeaders = req.Headers.ExactCaseItems()
+	}
 	// Inject tracing system baggage
-	reqHeaders := tchannel.InjectOutboundSpan(call.Response(), req.Headers.Items())
+	reqHeaders = tchannel.InjectOutboundSpan(call.Response(), reqHeaders)
 
-	if err := writeRequestHeaders(ctx, format, reqHeaders, call.Arg2Writer); err != nil {
+	if err := writeRequestHeaders(ctx, format, reqHeaders, call.Arg2Writer, o.transport.exactCaseHeader); err != nil {
```

Comment:
We've already decided which set of items we're using, I don't think we need to pass this option through?

Reply:
There was a comment from @abhinav here that we only to consolidate the check of writeExactHeaderCase at one place and that would be `writeRequeatHeaders`. We have channel_outbound.go and outbound.go here both using this `writeRequestHeaders` method. It is true that we already decided which set of request headers to use here but we still need this flag to decide whether to canonicalize the jaeger tracing headers. This is how it works in the old path.